zebra/zebra_evpn_mh.c

```diff
@@ -3922,6 +3922,13 @@ void zebra_evpn_proc_remote_nh(ZAPI_HANDLER_ARGS)
 	struct prefix_evpn dummy_prefix;
 
 	s = msg;
+
+	/*
+	 * Ensure that the stream sent to us is long enough
+	 */
+	if (hdr->length < (4 + sizeof(nh) + sizeof(rmac)))
```

Comment:
Not familiar with the code, but it looks like `rmac` is read only when the command is `ADD`.
So in case of `DEL` I suppose the length may be `4 + sizeof(nh)`.

Reply:
fixed